tensorboard/plugins/debugger_v2/debugger_v2_plugin_test.py

```diff
@@ -18,26 +18,107 @@
+        logdir, circular_buffer_size=-1
+    )
+    try:
+
+        @tf.function
+        def unstack_and_sum(x):
+            elements = tf.unstack(x)
+            return elements[0] + elements[1] + elements[2] + elements[3]
+
+        @tf.function
+        def repeated_add(x, times):
+            sum = tf.constant(0, dtype=x.dtype)
+            i = tf.constant(0, dtype=tf.int32)
+            while tf.less(i, times):
+                sum += x
+                i += 1
+            return sum
+
+        @tf.function
+        def my_function(x):
+            times = tf.constant(3, dtype=tf.int32)
+            return repeated_add(unstack_and_sum(x), times)
+
+        x = tf.constant([1, 3, 3, 7], dtype=tf.float32)
+        for i in range(3):
+            assert my_function(x).numpy() == 42.0
+        writer.FlushNonExecutionFiles()
+        writer.FlushExecutionFiles()
+    finally:
+        tf.debugging.experimental.disable_dump_debug_info()
+
+
+_ROUTE_PREFIX = "/data/plugin/debugger-v2"
+
+
+@test_util.run_v2_only("tfdbg2 is not available in r1.")
 class DebuggerV2PluginTest(tf.test.TestCase):
-    def testInstantiatePlugin(self):
-        dummy_logdir = tempfile.mkdtemp()
-        context = base_plugin.TBContext(logdir=dummy_logdir)
-        plugin = debugger_v2_plugin.DebuggerV2Plugin(context)
-        self.assertTrue(plugin)
+    def setUp(self):
+        super(DebuggerV2PluginTest, self).setUp()
+        self.logdir = tempfile.mkdtemp()
+        context = base_plugin.TBContext(logdir=self.logdir)
+        self.plugin = debugger_v2_plugin.DebuggerV2Plugin(context)
+        wsgi_app = application.TensorBoardWSGI([self.plugin])
+        self.server = werkzeug_test.Client(wsgi_app, wrappers.BaseResponse)
+
+    def tearDown(self):
+        if os.path.isdir(self.logdir):
+            shutil.rmtree(self.logdir)
+        super(DebuggerV2PluginTest, self).tearDown()
 
     def testPluginIsNotActiveByDefault(self):
-        dummy_logdir = tempfile.mkdtemp()
-        context = base_plugin.TBContext(logdir=dummy_logdir)
-        plugin = debugger_v2_plugin.DebuggerV2Plugin(context)
-        self.assertFalse(plugin.is_active())
+        self.assertFalse(self.plugin.is_active())
+
+    def testServeRunsWithoutExistingRuns(self):
+        response = self.server.get(_ROUTE_PREFIX + "/runs")
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(
+            "application/json", response.headers.get("content-type")
+        )
+        self.assertEqual(json.loads(response.get_data()), [])
+
+    def testServeRunsWithExistingRuns(self):
+        _generate_tfdbg_v2_data(self.logdir)
+        response = self.server.get(_ROUTE_PREFIX + "/runs")
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(
+            "application/json", response.headers.get("content-type")
+        )
+        self.assertEqual(
+            json.loads(response.get_data()), ["__default_debugger_run__"]
```

Comment:
Consider using `local_data_provider.DEFAULT_DEBUGGER_RUN_NAME` here to
avoid skew?